ui/src/widgets/ui-gauge/types/UIGaugeDial.vue

```diff
@@ -215,12 +230,29 @@ export default {
             // update sections
             this.updateSegmentArc()
 
+            const svgPaths = this.svg.select('#sections').selectAll('path')
+            const pathNodes = svgPaths._groups[0]
             const segments = this.segments
-            this.svg.select('#sections')
-                .selectAll('path')
-                .data(segments)
-                .enter()
-                .append('path')
+
+            // check there are more paths than segments
+            if (pathNodes.length !== 0 && pathNodes.length > segments.length) {
+                this.svg.select('#sections')
+                    .selectAll('path')
+                    .data(segments)
+                    .enter()
+                    .append('path')
+                    // remove any extra paths
+                    .filter((_d, i) => {
+                        return i > segments.length - 1
+                    })
+                    .remove()
+            } else {
+                this.svg.select('#sections')
+                    .selectAll('path')
+                    .data(segments)
+                    .enter()
+                    .append('path')
+            }
```

Comment:
force adding a gauge without any segments throws an error in the console, check line 260

Reply:
Checking this

Reply:
@cstns This is fixed with https://github.com/FlowFuse/node-red-dashboard/pull/1238/commits/d035a588fbc659550b1f255af263427441ff12bd